data/data/bootstrap/aro/files/systemd/units/mdsd.service

```diff
@@ -0,0 +1,27 @@
+[Unit]
+After=network-online.target
+StartLimitIntervalSec=0
+
+[Service]
+RestartSec=1s
+EnvironmentFile=/etc/sysconfig/mdsd
+ExecStartPre=-/bin/podman rm -f %N
+ExecStartPre=-mkdir /var/run/mdsd
+ExecStartPre=-/bin/podman pull $MDSDIMAGE
+ExecStart=/bin/podman run \
+  --entrypoint /usr/sbin/mdsd \
+  --net=host \
+  --hostname `hostname` \
```

Comment:
does this work in a systemd service file?

Reply:
This is a commit from the ARO installer fork. I assume it works.

Reply:
I mean, it definitely doesn't :slightly_smiling_face:

Reply:
@zaneb Can you elaborate?

Reply:
ExecStart commands don't run in a shell, they're passed to execv or whatever after some very light interpretation by systemd, which does not include command substitution. So this will pass the literal string `` `hostname` `` as the hostname argument to podman.

Reply:
Which in turn would be passed to the invoked mdsd command where it would be interpreted in the container shell, right? I've spoken to ARO and this works as intended.

Reply:
Good catch! We had a discussion and determined we don't need the `` --hostname `hostname` `` parameter at all, mdsd uses the node name from `MONITORING_ROLE_INSTANCE` in the env file. And indeed on the cluster nodes we run mdsd in a pod and set `MONITORING_ROLE_INSTANCE` to `{fieldPath: spec.nodeName}` and that works just fine.

I'll open a PR against the ARO fork to drop this parameter there as well.